guides/release-process.md

```diff
@@ -1,8 +1,8 @@
-# Deployment
+# Release Process
 
-These deployment procedures mainly concern the Cypress binary and `cypress` npm module.
+These procedures concern the release process for the Cypress binary and `cypress` npm module.
 
-Independent `@cypress/` packages that live inside the [`npm`](./npm) directory are automatically published to npm (with [`semantic-release`](https://semantic-release.gitbook.io/semantic-release/)) upon being merged into master. You can read more about this in [CONTRIBUTING.md](./CONTRIBUTING.md#committing-code)
+The `@cypress/`-namespaced NPM packages that live inside the [`/npm`](../npm) directory are automatically published to npm (with [`semantic-release`](https://semantic-release.gitbook.io/semantic-release/)) upon being merged into `master`. You can read more about this in [CONTRIBUTING.md](../CONTRIBUTING.md#independent-packages-ci-workflow).
 
 Anyone can build the binary and npm package, but you can only deploy the Cypress application and publish the npm module `cypress` if you are a member of the `cypress` npm organization.
```

Comment:
IMO it'd be worth adding a note to the Building Locally section that this section is useful for local debugging since CI handles the logic. When releasing, I read through each step to realize I could skip it all.

Reply:
Moved this to its own guide entirely, it's confusing to have it here at all.  6de8200